### ISSUE TITLE
Ingress chart

### ISSUE THREAD

**Issue by pierreozoux** [helm/charts#727]
I'm wondering if an `ingress` chart would make sense.

Let me expose the use case:

Here would be the file:
https://github.com/kubernetes/charts/blob/671108ad8889c07c3484e62217af62f592d95f77/stable/wordpress/templates/ingress.yaml

And then, we can make every web app charts require this chart, this would DRY things.

What do you think about the general idea?

We could use this [new tags and conditions feature](https://github.com/kubernetes/helm/blob/master/docs/charts.md#tags-and-condition-fields-in-requirementsyaml) to actually load the `ingress` chart or not.

This would also dry `NOTES` from `nginx-ingress` and `kube-lego`.

And finally this would help people bootstrap a secured infrastructure.

At the end, we would have this:

```
helm install --set tags.tls=enabled --set host=example.org stable/wordpress
```
And this would deploy a full tls WP!

**Comment by mgoodness** [comment]
@pierreozoux there's been discussion (offline, unfortunately) about having a "common" chart that would be a dependency of every other chart in this repo. The community would use that chart to apply consistent best practices and, as you suggest, provide common functionality.

TL;DR: good idea!

**Comment by pierreozoux** [comment]
Closing in favor of #794
